src/Microsoft.IdentityModel.Tokens/EncryptingCredentials.cs

```diff
@@ -25,67 +25,109 @@
         /// </summary>
         /// <param name="key"><see cref="SecurityKey"/></param>
-        /// <param name="alg">The key encryption algorithm to apply.</param>
-        /// <param name="enc">The encryption algorithm to apply.</param>
+        /// <param name="alg">A key wrap encryption algorithm to apply when encrypting a session key.</param>
+        /// <param name="enc">Data encryption algorithm to apply when encrypting plaintext.</param>
+        /// <exception cref="ArgumentNullException">if 'key' is null.</exception>
+        /// <exception cref="ArgumentNullException">if 'alg' is null or empty.</exception>
+        /// <exception cref="ArgumentNullException">if 'enc' is null or empty.</exception>
         public EncryptingCredentials(SecurityKey key, string alg, string enc)
         {
-            if (key == null)
-                throw LogHelper.LogArgumentNullException(nameof(key));
+            Key = key;
+            Alg = alg;
+            Enc = enc;
+        }
 
-            if (string.IsNullOrWhiteSpace(alg))
-                throw LogHelper.LogArgumentNullException(nameof(alg));
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EncryptingCredentials"/> class.
+        /// </summary>
+        /// <remarks> Used in scenarios when a key represents a 'shared' symmetric key.
+        /// For example, SAML 2.0 Assertion will be encrypted using a provided symmetric key
+        /// which won't be serialized to a SAML token.
+        /// </remarks>
+        /// <param name="key"><see cref="SecurityKey"/></param>
+        /// <param name="enc">Data encryption algorithm to apply when encrypting plaintext.</param>
+        /// <exception cref="ArgumentException">If the <see cref="SecurityKey"/> is not <see cref="SymmetricSecurityKey"/>.</exception>
+        /// <exception cref="ArgumentNullException">if 'enc' is null or empty.</exception>
+        public EncryptingCredentials(SecurityKey key, string enc)
+        {
+            Key = key;
 
-            if (string.IsNullOrWhiteSpace(enc))
-                throw LogHelper.LogArgumentNullException(nameof(enc));
+            if (key.GetType() != typeof(SymmetricSecurityKey))
+                throw LogHelper.LogArgumentException<ArgumentException>("key", LogMessages.IDX10704);
 
-            Alg = alg;
+            //explicitly setting Alg to None
+            Alg = SecurityAlgorithms.None;
             Enc = enc;
-            Key = key;
         }
 
         /// <summary>
-        /// Gets the algorithm which used for token encryption.
+        /// Gets the key wrap encryption algorithm used for a session key encryption.
         /// </summary>
         public string Alg
         {
-            get;
-            private set;
+            get => _alg;
+            private set => _alg = string.IsNullOrEmpty(value) ? throw LogHelper.LogArgumentNullException("alg") : value;
         }
```

Comment:
Maybe pass in nameof(value) instead of "alg" to LogArgumentNullException()?

Reply:
Not sure if it makes sense as the resulting log message would look something like this:
*IDX10000: The parameter **value** cannot be a 'null' or an empty object.*

Using reflection, we can get names of parameters from a constructor but I don't believe that is a good solution.